File: app/src/main/java/com/fittrack/data/entity/ExerciseLog.kt
package com.fittrack.data.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "exercise_logs",
    foreignKeys = [
        ForeignKey(
            entity = Exercise::class,
            parentColumns = ["id"],
            childColumns = ["exerciseId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index("exerciseId"), Index("workoutSessionId")]
)
data class ExerciseLog(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val exerciseId: Int,
    val routineId: Int,
    val userId: Int,
    val workoutSessionId: String,
    val setNumber: Int,
    val weight: Double,
    val reps: Int,
    val completedAt: Long = System.currentTimeMillis()
)

File: ios/P3Fitness/Models/ExerciseLog.swift
import SwiftData
import Foundation

@Model
final class ExerciseLog {
    var id: Int
    var exerciseId: Int
    var routineId: Int
    var userId: Int
    var workoutSessionId: String
    var setNumber: Int
    var weight: Double
    var reps: Int
    var completedAt: Date

    init(exerciseId: Int, routineId: Int, userId: Int, workoutSessionId: String, setNumber: Int, weight: Double, reps: Int) {
        self.id = Int(Date().timeIntervalSince1970 * 1000000) % 1_000_000_000
        self.exerciseId = exerciseId
        self.routineId = routineId
        self.userId = userId
        self.workoutSessionId = workoutSessionId
        self.setNumber = setNumber
        self.weight = weight
        self.reps = reps
        self.completedAt = Date()
    }
}
